<!--
  1 failure:  :disappointed_rel...
  0 warning: 
  1 messages
  
  DangerID: danger-id-default;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  :disappointed_relieved: `npm test` failed. See below for details.
  </td>
    </tr>
  </tbody>
</table>



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  ```
> titanium-mobile@7.4.0 test /Users/build/jenkins/workspace/dk_titanium_mobile_PR-10140-BWQNTU243W6VZJY6WXGQZZTFJJLOHU3UEKYXQLKVJ73XZ5IJPJ5A
> grunt

Running "appcJs:src:lintOnly" (appcJs) task

Running "eslint:src" (eslint) task

/Users/build/jenkins/workspace/dk_titanium_mobile_PR-10140-BWQNTU243W6VZJY6WXGQZZTFJJLOHU3UEKYXQLKVJ73XZ5IJPJ5A/tests/Resources/ti.ui.tableview.addontest.js
  11:5   warning  'should' is assigned a value but never used  no-unused-vars
  32:20  error    Missing space before value for key 'title'   key-spacing

✖ 2 problems (1 error, 1 warning)
  1 error, 0 warnings potentially fixable with the `--fix` option.

Warning: Task "eslint:src" failed. Use --force to continue.

Aborted due to warnings.
npm ERR! Test failed.  See above for more details.

```
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

--- a/iphone/Classes/TiUITableViewProxy.m
+++ b/iphone/Classes/TiUITableViewProxy.m
@@ -421,7 +421,7 @@ USE_VIEW_FOR_CONTENT_HEIGHT
     TiUITableViewAction *action = [[[TiUITableViewAction alloc] initWithObject:newrow animation:anim type:TiUITableViewActionUpdateRow] autorelease];
     [table dispatchAction:action];
   },
-      YES);
+      [NSThread isMainThread]);
 }
 
 - (void)deleteRow:(id)args
@@ -834,7 +834,7 @@ DEFINE_DEF_PROP(scrollsToTop, [NSNumber numberWithBool:YES]);
     }
     [oldSections release];
   },
-      NO);
+      [NSThread isMainThread]);
 }
 
 - (void)setSections:(NSArray *)newSections
@@ -1016,7 +1016,7 @@ DEFINE_DEF_PROP(scrollsToTop, [NSNumber numberWithBool:YES]);
         }
                             forceReload:NO];
   },
-      NO);
+      [NSThread isMainThread]);
 }
 
 - (void)insertSectionAfter:(id)args

--- a/tests/Resources/ti.ui.tableview.addontest.js
+++ b/tests/Resources/ti.ui.tableview.addontest.js
@@ -8,7 +8,6 @@
 /* global Ti */
 /* eslint no-unused-expressions: "off" */
 'use strict';
-var should = require('./utilities/assertions');
 
 describe('Titanium.UI.TableView', function () {
 	var win,
@@ -29,7 +28,7 @@ describe('Titanium.UI.TableView', function () {
 
 	it.ios('TIMOB-26164 : updateRow + insertRowAfter causing crash on main thread', function (finish) {
 		var tableView = Ti.UI.createTableView({
-			data: [ { title:'Red' } ]
+			data: [ { title: 'Red' } ]
 		});
 
 		win = Ti.UI.createWindow({

